src/test/java/api/exercise/Exercise1.java

```diff
@@ -0,0 +1,105 @@
+package api.exercise;
+
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@SuppressWarnings({"ConstantConditions", "unused", "MismatchedQueryAndUpdateOfCollection"})
+class Exercise1 {
+
+    enum Status {
+        UNKNOWN,
+        PENDING,
+        COMMENTED,
+        ACCEPTED,
+        DECLINED
+    }
+
+    @Test
+    void acceptGreaterThan21OthersDecline() {
+        Person alex = new Person("Алексей", "Мельников", 20);
+        Person ivan = new Person("Иван", "Стрельцов", 24);
+        Person helen = new Person("Елена", "Рощина", 22);
+        Map<Person, Status> candidates = new HashMap<>();
+        candidates.put(alex, Status.PENDING);
+        candidates.put(ivan, Status.PENDING);
+        candidates.put(helen, Status.PENDING);
+
+        candidates.replaceAll((person, status) -> person.getAge() > 21? Status.ACCEPTED : Status.DECLINED);
+
+        assertThat(candidates, Matchers.hasEntry(ivan, Status.ACCEPTED));
+        assertThat(candidates, Matchers.hasEntry(helen, Status.ACCEPTED));
+        assertThat(candidates, Matchers.hasEntry(alex, Status.DECLINED));
+    }
+
+    @Test
+    void acceptGreaterThan21OthersRemove() {
+        Person alex = new Person("Алексей", "Мельников", 20);
+        Person ivan = new Person("Иван", "Стрельцов", 24);
+        Person helen = new Person("Елена", "Рощина", 22);
+        Map<Person, Status> candidates = new HashMap<>();
+        candidates.put(alex, Status.PENDING);
+        candidates.put(ivan, Status.PENDING);
+        candidates.put(helen, Status.PENDING);
+
+        candidates.put(new Person("a", "a", 19), Status.PENDING);
+        candidates.put(new Person("b", "c", 16), Status.PENDING);
+        candidates.put(new Person("b", "c", 5), Status.PENDING);
+
+        candidates.replaceAll((person, status) -> person.getAge() > 21? Status.ACCEPTED : Status.DECLINED);
+        candidates.entrySet().removeIf(personStatusEntry -> personStatusEntry.getValue().equals(Status.DECLINED));
```

Comment:
You can optimize your solution by swapping these operations (removeIf first and then replaceAll), because now you're going through the whole map twice.